index.js

```diff
@@ -213,56 +216,55 @@ class AutocompletePrompt extends Base {
   }
 
   search(searchTerm /* : ?string */) {
-    var self = this;
-    self.selected = 0;
+    this.selected = 0;
 
     // Only render searching state after first time
-    if (self.searchedOnce) {
-      self.searching = true;
-      self.currentChoices = new Choices([]);
-      self.render(); // Now render current searching state
+    if (this.searchedOnce) {
+      this.searching = true;
+      this.currentChoices = new Choices([]);
+      this.render(); // Now render current searching state
     } else {
-      self.searchedOnce = true;
+      this.searchedOnce = true;
     }
 
-    self.lastSearchTerm = searchTerm;
+    this.lastSearchTerm = searchTerm;
 
-    var thisPromise;
+    let thisPromise;
     try {
-      var result = self.opt.source(self.answers, searchTerm);
+      const result = this.opt.source(this.answers, searchTerm);
       thisPromise = Promise.resolve(result);
     } catch (error) {
       thisPromise = Promise.reject(error);
     }
 
     // Store this promise for check in the callback
-    self.lastPromise = thisPromise;
+    const lastPromise = thisPromise;
 
-    return thisPromise.then(function inner(choices) {
+    return thisPromise.then((choices) => {
       // If another search is triggered before the current search finishes, don't set results
-      if (thisPromise !== self.lastPromise) return;
+      if (thisPromise !== lastPromise) return;
```

Comment:
Unsure if this check makes sense; it seems `lastPromise` isn't changed anywhere else?

Reply:
Makes perfect sense. Probably used it elsewhere before.